utils/hct/gen_intrin_main.txt

```diff
@@ -364,6 +364,9 @@ bool [[ro]] IsHelperLane();
 // HL Op for allocating ray query object that default constructor uses
 uint [[hidden]] AllocateRayQuery(in uint flags);
 
+// HL Op for allocating ray query object when non-zero 2nd template argument is given
+uint [[hidden]] AllocateRayQuery2(in uint flags, in uint rayqueryflags);
```

Comment:
I actually think we could have gotten away with using the same hidden intrinsic, with an argument update.  Then, the high level op is always the same, with two arguments, and the lowering just lowers to the prior DXIL op when the second arg is zero.